subt/main.py

```diff
@@ -889,6 +901,19 @@ def play_virtual_part_explore(self):
 
         self.stdout(self.time, "Explore phase finished %.3f" % dist, reason)
 
+    def play_virtual_part_map_and_explore_frontiers(self):
+        start_time = self.sim_time_sec
+        while self.sim_time_sec - start_time < self.timeout.total_seconds():
+            channel = self.update()
+            if channel == 'waypoints':
+                tmp_trace = Trace()
+                tmp_trace.trace = self.waypoints
+                self.waypoints = None
+                tmp_trace.reverse()
```

Comment:
Why? Is somebody sending commands in reverse order? Why?

Reply:
I am still not sure how to continue with `follow_trace`. If the waypoints are used from Freyja, then we do not care about Z coordinate (the link would not be connected, so it maybe does not matter that much). Yes, this reverse to match old "API" is strange :(.

Reply:
Let's discuss options on Wednesday.

Reply:
ACK